webapp/utils/post_utils.jsx

```diff
@@ -107,3 +107,20 @@ export function shouldShowDotMenu(post) {
 
     return false;
 }
+
+export function contains(text, key) {
+    if (!text || !key) {
+        return false;
+    }
+
+    const pattern = /`${key}[^\w-]`/;
+    function mapFn(t) {
+        return t === key || (t.startsWith(key) && pattern.test(t));
+    }
+
+    function reduceFn(a, b) {
+        return a || b;
+    }
+
+    return text.includes(key) && text.split(/[\s,.!:]/).map(mapFn).reduce(reduceFn, false);
```

Comment:
You can actually simplify this using `\b` in the regex. '\b' matches "word boundaries" which means the text changes between word characters (letters/numbers) and non-word characters (everything else). It's a bit tricky since the start of the string is a non-word character (`@`), but you can use a `\B` to detect that there's *not* a word boundary there and that the character before the @ isn't a letter. This pattern will only work when `key` doesn't contain a period or hyphen (we might want a comment to explain that), but we don't need to worry about those now.
```javascript
function containsAtMention(text, key) {
    if (!text || !key) {
        return false;
    }

    // This doesn't work for at mentions containing periods or hyphens
    return new RegExp(`\B${key}\b`, 'i').test(text);
}
```